pilot/pkg/security/authn/v1alpha1/policy_applier.go

```diff
@@ -331,28 +331,57 @@ func (a v1alpha1PolicyApplier) AuthNFilter(proxyType model.NodeType, isXDSMarsha
 	return out
 }
 
-func (a v1alpha1PolicyApplier) InboundFilterChain(sdsUdsPath string, meta *model.NodeMetadata) []plugin.FilterChain {
+func (a v1alpha1PolicyApplier) InboundFilterChain(sdsUdsPath string, node *model.Proxy) []plugin.FilterChain {
 	if a.policy == nil || len(a.policy.Peers) == 0 {
 		return nil
 	}
-	alpnIstioMatch := &ldsv2.FilterChainMatch{
-		ApplicationProtocols: util.ALPNInMesh,
-	}
-	tls := &auth.DownstreamTlsContext{
-		CommonTlsContext: &auth.CommonTlsContext{
-			// Note that in the PERMISSIVE mode, we match filter chain on "istio" ALPN,
-			// which is used to differentiate between service mesh and legacy traffic.
-			//
-			// Client sidecar outbound cluster's TLSContext.ALPN must include "istio".
-			//
-			// Server sidecar filter chain's FilterChainMatch.ApplicationProtocols must
-			// include "istio" for the secure traffic, but its TLSContext.ALPN must not
-			// include "istio", which would interfere with negotiation of the underlying
-			// protocol, e.g. HTTP/2.
-			AlpnProtocols: util.ALPNHttp,
-		},
-		RequireClientCertificate: protovalue.BoolTrue,
+	meta := node.Metadata
+	var alpnIstioMatch *ldsv2.FilterChainMatch
+	var tls *auth.DownstreamTlsContext
+	if util.IsTCPMetadataExchangeEnabled(node) {
+		alpnIstioMatch = &ldsv2.FilterChainMatch{
+			ApplicationProtocols: util.ALPNInMeshWithMxc,
+		}
+		tls = &auth.DownstreamTlsContext{
```

Comment:
there seems quite some duplicate in this if/else, at least for comments.

Reply:
Fixed